pkg/types/defaults/installconfig.go

```diff
@@ -129,4 +134,32 @@ func SetInstallConfigDefaults(c *types.InstallConfig) {
 	if c.AdditionalTrustBundlePolicy == "" {
 		c.AdditionalTrustBundlePolicy = types.PolicyProxyOnly
 	}
+
+	setPullSecretDefault(c)
+}
+
+func setPullSecretDefault(config *types.InstallConfig) {
+	if config.PullSecret != "" {
+		return
+	}
+
+	var pullSecretFile string
+	if pullSecretFile = os.Getenv("REGISTRY_AUTH_FILE"); pullSecretFile == "" {
```

Comment:
1. All of our envars are prefixed with `OPENSHIFT_INSTALL` so this should be `OPENSHIFT_INSTALL_REGISTRY_AUTH_FILE` 
2. The installer generally does not allow users to pass configuration through envvars. Is this something that we expect users to set or to only be used internally (CI)? If we expect users to set it, can we use a flag instead to get the same effect?

Reply:
A CLI flag seems to me like the way to go to allow users to make use of this. Env vars are not very discoverable.